lib/diagnostic.js

```diff
@@ -5,17 +5,27 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (originalArray, callback, initalValue) {
+  const workingArray = originalArray.slice();
```

Comment:
Due to this function's arguments being assumed to be two numbers, ```originalArray``` is assumed to be a number and since you are trying to call ```slice()```, an array method, on a number, this will throw an error.